modules/javafx.controls/src/main/java/javafx/scene/control/skin/MenuBarSkin.java

```diff
@@ -733,8 +730,8 @@ public void dispose() {
      *                                                                         *
      **************************************************************************/
 
-    // For testing purpose only.
-    MenuButton getNodeForMenu(int i) {
+    // package protected for testing purposes
+    MenuBarButton menuBarButtonAt(int i) {
         if (i < container.getChildren().size()) {
```

Comment:
This method throws `IndexOutOfBoundsException` if the index is negative, but returns `null` if the index is larger than the collection size. None of the callers account for this, so we'd just get `NullPointerException` at the call sites.

Since you touched this method, I suggest replacing the implementation with
```java
MenuBarButton menuBarButtonAt(int i) {
    return (MenuBarButton)container.getChildren().get(i);
}
```

Reply:
yes, the suggested change is safe: the product code always guards against invalid indexes, while the test code always expects a non-null return value.

the only time where it might backfire is if a (new) test sets up an empty menu bar.  but we don't have such a test.